go/libraries/doltcore/doltdb/doltdb.go

```diff
@@ -1246,6 +1246,56 @@ func (ddb *DoltDB) NewBranchAtCommit(ctx context.Context, branchRef ref.DoltRef,
 	return ddb.UpdateWorkingSet(ctx, wsRef, ws, currWsHash, TodoWorkingSetMeta(), replicationStatus)
 }
 
+func (ddb *DoltDB) NewBranchAtCommitWithWorkingSet(ctx context.Context, branchRef ref.DoltRef, commit *Commit, ws *WorkingSet, replicationStatus *ReplicationStatusController) error {
```

Comment:
It's kind of silly to introduce this new operation when it's only used for rebase. Better to have two separate operations: create the branch, then set its working set.

Reply:
I was trying to avoid resolving the commit multiple times and assigned the working set multiple times.
Are you saying `rebaseRefs()` should call `NewBranchAtCommit()` and then `UpdateWorkingSet()` afterwards?

Reply:
Yup, use the existing doltdb methods instead of a new one.